lsp/gopls.lua

```diff
@@ -4,13 +4,82 @@
 ---
 --- Google's lsp server for golang.
 
+--- @class go_dir_custom_args
+---
+--- @field envvar_id string
+---
+--- @field custom_subdir string?
+
 local mod_cache = nil
+local std_lib = nil
+
+---@param custom_args go_dir_custom_args
+---@param on_complete fun(dir: string | nil)
+local function identify_go_dir(custom_args, on_complete)
+  local cmd = { 'go', 'env', custom_args.envvar_id }
+  vim.system(cmd, { text = true }, function(output)
+    if output.code == 0 then
+      if output.stdout then
```

Comment:
what happens if `output.stdout` is nil? should it be like this : 
```suggestion
    if output.code == 0 and output.stdout then
```

Reply:
yes, the nested if can be put together. `GOMODCACHE` and `GOROOT` will always be set in a system where go is installed.

So the only situation I can think of where `output.stdout` is empty would be in a system without go installed.

Though, on my system `output.stdout` is never `nil`, its just an empty string when I pass in a non-existing env var, e.g. `GOBLABLA`. I suggest to rewrite to

```
    if output.code == 0 and output.stdout and output.stdout ~= "" then
```

It is a bit verbose but catches all non-happy paths.

Reply:
applied.